libyul/backends/wasm/WasmCodeTransform.cpp

```diff
@@ -400,6 +398,23 @@ void WasmCodeTransform::allocateGlobals(size_t _amount)
 {
 	while (m_globalVariables.size() < _amount)
 		m_globalVariables.emplace_back(wasm::GlobalVariableDeclaration{
-			m_nameDispenser.newName("global_"_yulstring).str()
+			m_nameDispenser.newName("global_"_yulstring).str(),
+			"i64"
 		});
 }
+
+wasm::Literal WasmCodeTransform::makeLiteral(wasm::Type _type, u256 _value)
```

Comment:
Well you said PR is not changing the code transform, perhaps could have left out this part :)

Reply:
I only said that the generated code does not change and that the use of `i32` is not enabled :P But the code for handling `i32` is already there. It's more prominent in `BinaryTransform`. Here I only needed this one function.

I know, it's not a perfect way to split this. Due to `i64` being hard-coded, the `i32` branches can never get executed in this PR. But I wanted to reduce the size of #8240 and this, not-quite-natural split was the only way I could come up with to get a coherent PR.

Reply:
Though, yeah, maybe it would be just fine without it. I'll move it.

Reply:
Done. This is now in #8240.